File: template/prime-number/sieve-of-eratosthenes.js
function findPrimes(targetNumber) {
  // 소수인지 확인하는 배열, 인덱스는 숫자고, 0, 1은 아님
  const isPrimes =  [ false, false, ...Array(targetNumber - 1).fill(true)];
  const primes = [];
  const squareRoot = Math.sqrt(targetNumber) 

  // 2에서 타겟 숫자의 제곱근까지의 모든 숫자를 전부 순회
  for (let number = 2; number <= squareRoot; number++) {
    // 해당 숫자가 소수면, 그 숫자의 배수들을 모두 소수가 아니라고 표시
    if (isPrimes[number]) {
      for (let multiple = number * number; multiple <= targetNumber; multiple += number) {
        isPrimes[multiple] = false;
      }
    }
  }
  // 소수인지 확인하는 배열을 전부 둘면서 true면 소수에 담는다. 
  for (const [index, isPrime] of isPrimes.entries()) {
    if (isPrime) {
      primes.push(index);
    }
  }
  return primes;
}

File: template/prime-number/sieve-of-eratosthenes.py
def sieve_of_eratosthenes(target_number):
    # n만큼의 배열을 생성
    isPrimes = [True] * (target_number+1)
    # 0, 1은 False, 소수가 아님
    isPrimes[0], isPrimes[1] = [False] * 2
    primes = []
    
    square_root = int(target_number**(0.5))
    
    # 2에서 n의 제곱근까지 모든 수를 순회
    for number in range(2, square_root + 1):
        # 그 수가 소수라고 표시가 되어 있다면, 
        if isPrimes[number]:
            # 그 수의 제곱 부터 n까지의 숫자 중, 수의 배수를 제외
            for i in range(number*number, target_number+1, number):
                isPrimes[i] = False
    
    # 배열의 모든 수를 순회 하면서 소수면 추가.
    for number, isPrime in enumerate(primes):
        if isPrime:
            primes.append(number)

    return primes
